### DISCUSSION TITLE
Issue with "Delete" button (Chrome) and resources list page (Firefox)  on production server

### DISCUSSION THREAD

**Discussion by babakmomeni** [Laravel-Backpack/community-forum#547] [General] [answered]
Hi folks
I've experienced an issue with BackPack for Laravel on my production server. Need to mention that everything working fine on local development environment. Following you can find description of issue, version and infrastructure information:

Issue:
------
1. Chrome: for all resources "Delete" button does not work at all and Dev console log shows an error as followings:

    Uncaught ReferenceError: deleteEntry is not defined
    at HTMLAnchorElement.onclick (prod-category:947:9)
onclick @ prod-category:947

<img width="1280" alt="Screen Shot 1402-04-09 at 14 47 46" src="https://github.com/Laravel-Backpack/community-forum/assets/42409425/53eb8ba2-7418-4f72-9a2f-6abaaee29e24">

2. Firefox: All resorce pages show an alert notification two times and then gray popup of 419 page expired error. notification windows contains : 
DataTables warning: table id=crudTable - Ajax error. For more information about this error, please see http://datatables.net/tn/7
<img width="904" alt="Screen Shot 1402-04-09 at 14 51 19" src="https://github.com/Laravel-Backpack/community-forum/assets/42409425/541c6fa4-4930-41ee-9df8-4e83687f6f31">
and then:
<img width="1267" alt="Screen Shot 1402-04-09 at 14 52 14" src="https://github.com/Laravel-Backpack/community-forum/assets/42409425/8d7342e2-b24e-4129-a075-6c021037a9fc">

Server:
-------
CentOS 7 with Docker CE latest version installed
BackPack installed on a container with latest version of Laravel (10 for time of issuing the note)
NGINX container as reverse proxy for handling input requests
SSL from Letsencrypt issued by our DNS service provider

Laravel and BackPack (plus PRO license) Version info:
------------------------------------------------------
As per output of artisan command of php artisan backpack:version

### PHP VERSION:
PHP 8.1.19 (cli) (built: May 24 2023 15:07:34) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.19, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.19, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v10.13.5@03106ae9ba2ec4b36dc973b7bdca6fad81e032b4

### BACKPACK PACKAGE VERSIONS:
backpack/crud: 5.6.1
backpack/filemanager: 2.0.1
backpack/generators: 3.3.16
backpack/pro: 1.6.6
 
I have searched and try some solutions from discussions with no result. Can anyone kindly help to solve the problem?

**Comment by jcastroa87** [comment]
Hello @babakmomeni 

Did you migrate vendor to server or installed on server? if you migrate, did you try to run?
```
composer dump-autoload
```

try to clear everything:
```
php artisan optimize:clear
```

Did the delete is hard or soft?

And finally, did you check laravel logs to get more information about expired?

Cheers.

**Reply by babakmomeni** [reply]
Hi @jorgetwgroup 
Thanks for your prompt reply
I'd done both, migrate and install. results are same.
I did artisan commands you wrote. Unfortunately none of them worked. 
Since the problem seems to be Ajax and nothing transmit to  Laravel so runtime logs do not contain relevant information...

**Comment by jcastroa87** [comment]
In any case we need more details, per example if firefox show thats errors is because receive bad response from backend so in network logs you will be able to get error details, or if this is a 500 error in laravel logs will se details, please remember no enable "APP_DEBUG" in .env

Thanks

**Comment by babakmomeni** [comment]
@jorgetwgroup 
For Firefox all requests are 200 (OK) except one results in 419. Detailed Network logs for its header, request and response are as followings:

Headers
-----------
<html><head></head><body><div class="headers-overview"><div class="url-preview"><div class="properties-view"><div class="tree-container">

POST | https://panel.polluxteam.ir/admin/product/search
-- | --


</body></html>POST
	https://panel.polluxteam.ir/admin/product/search
Status
419
No Reason Phrase
VersionHTTP/2
Transferred11.42 kB (10.67 kB size)
Referrer Policystrict-origin-when-cross-origin

	
cache-control
	no-cache, private
content-type
	application/json
date
	Fri, 30 Jun 2023 15:08:47 GMT
server
	ArvanCloud
server-timing
	total;dur=308
set-cookie
	pollux_session=eyJpdiI6Ik9jQTVqeDdmclFQaWJvbm1FQ1E2U1E9PSIsInZhbHVlIjoiVEYwV2lWN3BIMWt4YUtwWVB5dnFRWDMvUHlqNkZHcGp2THQ2cmVYc3YzbnF4Z0UwbHRSeWJjeTRweHNBL2QzOG9mNXJEcVRiZ0dzQ2VWcWY2ZWpVSkZkbk4rMUlnR1dBc0tiTVBBbjh3NUNNL0FVVHZkbm9EcmlsNS9uUEZIZzciLCJtYWMiOiI5ODdkOWUyOTg2MTM1M2NmM2ExMzBjM2I4ZWUyYjQ5NWE4YmUwMGI3NjJjZGNkYWYxOWNkZWJhNTkzZjliODVkIiwidGFnIjoiIn0%3D; expires=Fri, 30 Jun 2023 17:08:47 GMT; Max-Age=7200; path=/; domain=panel.polluxteam.ir; httponly; samesite=lax
X-Firefox-Spdy
	h2
x-request-id
	1064d8cd043ebe79a967426b0c501db0
x-sid
	4103
	
Accept
	application/json, text/javascript, */*; q=0.01
Accept-Encoding
	gzip, deflate, br
Accept-Language
	en-US,en;q=0.5
Connection
	keep-alive
Content-Length
	914
Content-Type
	application/x-www-form-urlencoded; charset=UTF-8
Host
	panel.polluxteam.ir
Origin
	https://panel.polluxteam.ir/
Sec-Fetch-Dest
	empty
Sec-Fetch-Mode
	cors
Sec-Fetch-Site
	same-origin
TE
	trailers
User-Agent
	Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/114.0
X-CSRF-TOKEN
	k64IVPnIZvcrJFymNNv3jixdbFeMMeXn5apScIam

Request
----------
{
	"draw": "1",
	"columns[0][data]": "0",
	"columns[0][name]": "",
	"columns[0][searchable]": "true",
	"columns[0][orderable]": "true",
	"columns[0][search][value]": "",
	"columns[0][search][regex]": "false",
	"columns[1][data]": "1",
	"columns[1][name]": "",
	"columns[1][searchable]": "true",
	"columns[1][orderable]": "true",
	"columns[1][search][value]": "",
	"columns[1][search][regex]": "false",
	"columns[2][data]": "2",
	"columns[2][name]": "",
	"columns[2][searchable]": "true",
	"columns[2][orderable]": "false",
	"columns[2][search][value]": "",
	"columns[2][search][regex]": "false",
	"columns[3][data]": "3",
	"columns[3][name]": "",
	"columns[3][searchable]": "true",
	"columns[3][orderable]": "false",
	"columns[3][search][value]": "",
	"columns[3][search][regex]": "false",
	"start": "0",
	"length": "10",
	"search[value]": "",
	"search[regex]": "false",
	"totalEntryCount": ""
}

Response
------------
{
	"message": "CSRF token mismatch.",
	"exception": "Symfony\\Component\\HttpKernel\\Exception\\HttpException",
	"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php",
	"line": 428,
	"trace": [
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php",
			"line": 399,
			"function": "prepareException",
			"class": "Illuminate\\Foundation\\Exceptions\\Handler",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
			"line": 51,
			"function": "render",
			"class": "Illuminate\\Foundation\\Exceptions\\Handler",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 185,
			"function": "handleException",
			"class": "Illuminate\\Routing\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php",
			"line": 49,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\View\\Middleware\\ShareErrorsFromSession",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php",
			"line": 121,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php",
			"line": 64,
			"function": "handleStatefulRequest",
			"class": "Illuminate\\Session\\Middleware\\StartSession",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Session\\Middleware\\StartSession",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php",
			"line": 37,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php",
			"line": 67,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Cookie\\Middleware\\EncryptCookies",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 116,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
			"line": 800,
			"function": "then",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
			"line": 777,
			"function": "runRouteWithinStack",
			"class": "Illuminate\\Routing\\Router",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
			"line": 741,
			"function": "runRoute",
			"class": "Illuminate\\Routing\\Router",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
			"line": 730,
			"function": "dispatchToRoute",
			"class": "Illuminate\\Routing\\Router",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
			"line": 200,
			"function": "dispatch",
			"class": "Illuminate\\Routing\\Router",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 141,
			"function": "Illuminate\\Foundation\\Http\\{closure}",
			"class": "Illuminate\\Foundation\\Http\\Kernel",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php",
			"line": 21,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php",
			"line": 31,
			"function": "handle",
			"class": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php",
			"line": 21,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php",
			"line": 40,
			"function": "handle",
			"class": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Foundation\\Http\\Middleware\\TrimStrings",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php",
			"line": 27,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php",
			"line": 86,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php",
			"line": 49,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Http\\Middleware\\HandleCors",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php",
			"line": 39,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 180,
			"function": "handle",
			"class": "Illuminate\\Http\\Middleware\\TrustProxies",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
			"line": 116,
			"function": "Illuminate\\Pipeline\\{closure}",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
			"line": 175,
			"function": "then",
			"class": "Illuminate\\Pipeline\\Pipeline",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
			"line": 144,
			"function": "sendRequestThroughRouter",
			"class": "Illuminate\\Foundation\\Http\\Kernel",
			"type": "->"
		},
		{
			"file": "/app/public/index.php",
			"line": 52,
			"function": "handle",
			"class": "Illuminate\\Foundation\\Http\\Kernel",
			"type": "->"
		},
		{
			"file": "/app/vendor/laravel/framework/src/Illuminate/Foundation/resources/server.php",
			"line": 16,
			"function": "require_once"
		}
	]
}

**Comment by jcastroa87** [comment]
You didnt customize the view and delete "<meta name="csrf-token" content="{{ csrf_token() }}" />" ??

Is look like server didnt receive "X-CSRF-TOKEN"

**Reply by babakmomeni** [reply]
Neither view nor delete button are not customized....

**Reply by babakmomeni** [reply]
I also, based on some older discussion, have tried to copy delete button from vendor to resource (technically making a custom button) and add CSRF token to jquery ajax request manually. No success...
of course, the change restored to original state and there is no custom delete button anymore...

**Comment by babakmomeni** [comment]
Another issue I've recently realized is in **Chrome**, multiple selection and then bulk delete shows a notty popup 
<img width="396" alt="Screen Shot 1402-04-09 at 19 02 49" src="https://github.com/Laravel-Backpack/community-forum/assets/42409425/64e41677-87fd-4abc-87f6-4e58e19deea2">

is this handful to determine the problem??

**Comment by jcastroa87** [comment]
This fully look as a server configuration problem.

**Reply by babakmomeni** [reply]
Can you give me a clue to find the problem?

**Reply by jcastroa87** [reply]
You can try to override function on your controller to see if headers are received.

Cheers.

**Reply by babakmomeni** [reply]
Could you please give more details or an example to do this?

**Comment by babakmomeni** [comment, marked as answer]
Dear @jorgetwgroup 
First of all thanks for your time and efforts to guide me through the problem
For future uses, I put the solution that worked for me.
Just add following code inside app/Providers/AppServiceProviders into register method and the issue is gone!

        if($this->app->environment('production')) {
            \URL::forceScheme('https');
        }

**Reply by jcastroa87** [reply]
Well done you find the problem, and thank to share.

**Reply by tabacitu** [reply]
Omg it was just a HTTPS problem 🤦‍♂️ Damn. Cheers to you guys for figuring it out 🍺
